### PR TITLE
[AudioToolbox] Add support for xcode 14 beta6.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

File: src/AudioUnit/AudioComponent.cs [hunk 1]
```diff
@@ -523,6 +523,128 @@ public AudioUnit CreateAudioUnit ()
...
+		[UnsupportedOSPlatform ("tvos")]
+#else
+		[NoWatch]
+		[NoTV]
+		[Mac (13,0)]
+		[iOS (16,0)]
+		[MacCatalyst (16,0)]
+#endif
+		public AudioComponentValidationResult Validate (NSDictionary validationParameters) {
+			var success = AudioComponentValidate (GetCheckedHandle (), validationParameters.GetNonNullHandle (nameof (validationParameters)), out var result);
```
Comment (rolfbjarne) [inline src/AudioUnit/AudioComponent.cs:591]: The header says `validationParameters` can be null.

File: src/AudioUnit/AudioComponent.cs [hunk 2]
```diff
@@ -523,6 +523,128 @@ public AudioUnit CreateAudioUnit ()
...
+#endif
+		public void ValidateAsync (NSDictionary validationParameters,
+				Action<AudioComponentValidationResult, NSDictionary?> onCompletion) {
+			if (onCompletion is null)
+				ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (onCompletion));
+			
+			var block_handler= new BlockLiteral ();
+			block_handler.SetupBlockUnsafe (static_action, onCompletion);
+			try {
+				AudioComponentValidateWithResults (GetCheckedHandle (), validationParameters.GetNonNullHandle (nameof (validationParameters)), ref block_handler);
```
Comment (rolfbjarne) [inline src/AudioUnit/AudioComponent.cs:642]: Same, headers say `validationParameters` can be null.

File: src/AudioUnit/AudioComponent.cs [hunk 3]
```diff
@@ -523,6 +523,128 @@ public AudioUnit CreateAudioUnit ()
...
+		[NoTV]
+		[Mac (13,0)]
+		[iOS (16,0)]
+		[MacCatalyst (16,0)]
+#endif
+		public AudioComponentValidationResult Validate (NSDictionary validationParameters) {
+			var success = AudioComponentValidate (GetCheckedHandle (), validationParameters.GetNonNullHandle (nameof (validationParameters)), out var result);
+			if (success == 0)
+				return result;
+			return AudioComponentValidationResult.Unknown;
```
Comment (rolfbjarne) [inline src/AudioUnit/AudioComponent.cs:594]: Your managed bindings have one thing in common: they don't expose the returned OSStatus from the native functions. This return value can be valuable when trying to diagnose why calls to these methods fail, so we often provide another overload that users can use to get this value. In this particular case that could be something like:

```cs
public AudioComponentValidationResult Validate (NSDictionary validationParameters, out int resultCode)
{
    ...
}
```

This goes for all the new API.

**Review by rolfbjarne — CHANGES_REQUESTED** [review]
Manual bindings need manual tests too :)

File: src/AudioToolbox/Enums.cs [hunk 4]
```diff
@@ -49,4 +49,18 @@ public enum AUSpatialMixerSourceMode : uint
...
+	[iOS (16, 0)]
+	public enum AUSpatialMixerPersonalizedHrtfMode : uint
+	{
+		[NoiOS, NoTV]
+		Off = 0,
+		[NoiOS, NoTV]
+		On = 1,
+		[NoiOS, NoTV]
+		Auto = 2,
+	}
```
Comment (tj-devel709) [inline src/AudioToolbox/Enums.cs:64]: Hmm why add tv and ios support if none of the enums support it?

**Review by tj-devel709 — COMMENTED** [review]
Other than this, LGTM

File: src/AudioUnit/AudioComponent.cs [hunk 5]
```diff
@@ -523,6 +523,156 @@ public AudioUnit CreateAudioUnit ()
...
+		[UnsupportedOSPlatform ("tvos")]
+#else
+		[NoWatch]
+		[NoTV]
+		[Mac (13,0)]
+		[iOS (16,0)]
+#endif
+		public NSDictionary? ConfigurationInfo {
+			get {
+				var success = AudioComponentCopyConfigurationInfo (GetCheckedHandle (), out var dictPtr);
```
Comment (rolfbjarne) [inline src/AudioUnit/AudioComponent.cs:554]: This one is also needs an overload (or really a method, since this is a property)

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: tests/monotouch-test/AudioToolbox/AudioComponentTest.cs [hunk 6]
```diff
@@ -117,6 +118,79 @@ public void TestResourceUsageInfoMachLookUpGlobalName ()
...
+			TestRuntime.AssertXcodeVersion (14, 0);
+			var resources = new ResourceUsageInfo ();
+			resources.IOKitUserClient = new string[] { "CustomUserClient1" };
+			resources.MachLookUpGlobalName = new string[] { "MachServiceName1" };
+			resources.NetworkClient = false;
+			resources.TemporaryExceptionReadWrite = false;
+
+			var componentInfo = new AudioComponentInfo ();
+			componentInfo.Type = AudioTypeOutput.Generic.ToString ();
+			componentInfo.Subtype =  "XMPL";
```
Comment (stephen-hawley) [inline tests/monotouch-test/AudioToolbox/AudioComponentTest.cs:134]: nit: extra space before `"XMPL"`

File: tests/monotouch-test/AudioToolbox/AudioComponentTest.cs [hunk 7]
```diff
@@ -117,6 +118,79 @@ public void TestResourceUsageInfoMachLookUpGlobalName ()
...
```
Comment (stephen-hawley) [inline tests/monotouch-test/AudioToolbox/AudioComponentTest.cs:157]: same

File: tests/monotouch-test/AudioToolbox/AudioComponentTest.cs [hunk 8]
```diff
@@ -117,6 +118,79 @@ public void TestResourceUsageInfoMachLookUpGlobalName ()
...
```
Comment (stephen-hawley) [inline tests/monotouch-test/AudioToolbox/AudioComponentTest.cs:180]: same

**Review by stephen-hawley — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**